projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -1,108 +1,190 @@
-        member_id: req.member_id,
-        "joining_user.login": `${req.joining_user.first_name || ""} ${
-          req.joining_user.last_name || ""
-        } (${req.joining_user.login || ""})`,
-        "grade": req.joining_user && req.joining_user.grade ? req.joining_user.grade : null,
-        at: req.at,
+    if (succ.false === undefined || succ.false === 0) {
+      this.messageService.add({
+        severity: "success",
+        summary: "Success",
+        detail: `${succ.true} request(s) have been ${msg}`,
+        life: TOAST_LENGTH,
+      });
+    } else if (succ.true === undefined || succ.true === 0) {
+      this.messageService.add({
+        severity: "error",
+        summary: "Error",
+        detail: `Unable to ${verb} the selected request(s).`,
+        life: TOAST_LENGTH,
+      });
+    } else {
+      this.messageService.add({
+        severity: "warn",
+        summary: "Partial success",
+        detail: `${succ.true} request(s) have been ${msg}, ${succ.false} could not be executed`,
+        life: TOAST_LENGTH,
       });
     }
   }
 
-  constructor(private groupService: GroupService) {}
+  _processRequestError(_err) {
+    this.messageService.add({
+      severity: "error",
+      summary: "Error",
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
+  }
+
+  isAccepting() {
+    return this.onGoingActivity === Activity.Accepting;
+  }
+
+  isRejecting() {
+    return this.onGoingActivity === Activity.Rejecting;
+  }
+
+  isIdle() {
+    return this.onGoingActivity === Activity.None;
+  }
+
+  constructor(
+    private groupService: GroupService,
+    private messageService: MessageService
+  ) {}
 
   ngOnInit() {
     this.panel.push({
       name: "Pending Requests",
       columns: this.columns,
     });
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
+  ngOnChanges(_changes: SimpleChanges) {
+    this.selection = [];
+    this._reloadData();
   }
 
-  onExpandWidth(e) {}
+  onAcceptOrReject(action: Action) {
+    if (this.selection.length === 0 || this.onGoingActivity !== Activity.None) {
+      return;
+    }
+
+    let resultObserver: Observable<RequestActionResponse>;
+    this.onGoingActivity = (action === Action.Accept) ? Activity.Accepting : Activity.Rejecting;
+    
+    const group_ids = this.selection.map((req: PendingRequest) => req.joining_user.group_id);
+
+    if (action === Action.Accept) {
+      resultObserver = this.groupService.acceptJoinRequest(this.id, group_ids);
+    } else {
+      resultObserver = this.groupService.rejectJoinRequest(this.id, group_ids);
+    }
 
-  onClickAccept(e) {
-    this.onAccept.emit(e);
+    resultObserver
+    .pipe(
```

Comment:
Cfr remark on moving it to the services and not checking for object but just having a correct interface.

Reply:
fixed, but not sure how to check the correct interface.

Reply:
But the question to ask yourself mainly is what happens if data is a .. let's say array instead of dict? Does it fail properly, does it fail at the place you want to? You should be able immediately, otherwise there is a problem. 
This is where unit tests would be very useful (and faster to re-test each time manually). We'll see later.
Actually I thought it would be handled by TS, but maybe it is not.. I'll check how this is supposed to work in TS ...

Reply:
Ok, there isn't a simple and good way to validate input to fail properly in JS/TS/NG.  🤷‍♂️ Ok we may come back on that later.